doc/topics/windows/windows-package-manager.rst

```diff
@@ -151,30 +151,100 @@ Maintaining Windows Repo Definitions in Git Repositories
+``salt-winrepo-ng`` is for newer minions (2015.8.0 and newer).
+
+Each software definition file contains all the information salt needs to install
+that software on a minion including the HTTP or ftp locations of the installer
+files, required command line switches for silent install, etc. Anyone is welcome
+to send a pull request to this repo to add new package definitions. The repos
+can be browsed here:
+`salt-winrepo`_
+`salt-winrepo-ng`_
+
+.. _salt-winrepo: https://github.com/saltstack/salt-winrepo.git
+.. _salt-winrepo-ng: https://github.com/saltstack/salt-winrepo-ng.git
 
-This repo points to the HTTP or ftp locations of the installer files. Anyone is
-welcome to send a pull request to this repo to add new package definitions.
-Browse the repo here: https://github.com/saltstack/salt-winrepo-ng.git
+.. note::
+    The newer software definition files are run through the salt's parser which
+    allows for the use of jinja.
 
 Configure which git repositories the master can search for package definitions
-by modifying or extending the :conf_master:`winrepo_remotes` option (**NOTE:**
-this option was called ``win_gitrepos`` in Salt versions earlier than 2015.8.0).
+by modifying or extending the :conf_master:`winrepo_remotes` and
+:conf_master:`winrepo_remotes_ng` options.
+
+.. important::
+    ``winrepo_remotes`` was called ``win_gitrepos`` in Salt versions earlier
+    than 2015.8.0
 
-Use the :mod:`winrepo.update_git_repos
-<salt.runners.winrepo.update_git_repos>` runner to clone/update the configured
-repos, then use :mod:`winrepo.genrepo <salt.runners.winrepo.genrepo>`
-runner to compile the repository cache. Finally, use :py:func:`pkg.refresh_db
-<salt.modules.win_pkg.refresh_db>` on each minion to have them update their
-copy of the repository cache. Command examples are as follows:
+Package definitions are pulled down from the online repository by running the
+:mod:`winrepo.update_git_repos <salt.runners.winrepo.update_git_repos>` runner.
+This command is run on the master:
 
 .. code-block:: bash
 
     salt-run winrepo.update_git_repos
+
+This will pull down the software definition files for older minions
+(``salt-winrepo``) and new minions (``salt-winrepo-ng``). They are stored in the
+``file_roots`` under ``win/repo/salt-winrepo`` and
+``win/repo-ng/salt-winrepo-ng`` respectively.
+
+.. important::
+    If you have customized software definition files that aren't maintained in a
+    repository, those should be stored under ``win/repo`` for older minions and
+    ``win/repo-ng`` for newer minions. The reason for this is that the contents
+    of ``win/repo/salt-winrepo`` and ``win/repo-ng/salt-winrepo-ng`` are wiped
+    out every time you run a ``winrepo.update_git_repos``.
+
+    Additionally, when you run ``winrepo.genrepo`` and ``pkg.refresh_db`` the
+    entire contents under ``win/repo`` and ``win/repo-ng``, to include all
+    subdirectories, are used to create the message pack file.
```

Comment:
If you are referring to `MessagePack`, it needs to be spelled with camel-case unless you are referring specifically to the Python implementation, which is `msgpack`.